docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,96 @@
+# Charm architecture
+
+The tmate-ssh-server-operator operates [tmate-ssh-server](https://github.com/tmate-io/tmate-ssh-server),
+which is the server side of [tmate](https://tmate.io/), an open-source terminal multiplexer
+that provides instant terminal sharing capabilities.
+The tmate-ssh-server application is built from source as an OCI image and runs as a systemd service on the machine.
+
+The charm provides the necessary connection details for a tmate client to connect to the tmate-ssh-server service
+via integration data.
+
+
+```mermaid
+C4Container
+title Container diagram for Tmate SSH Server
+
+
+ Container_Boundary(c1, "Tmate SSH Server") {
+    Container(tmatessh, "Tmate SSH Server", "", "Provides sharing a remote session")
+ }
+Container_Boundary(c2, "Tmate Client Charm") {
+    Container(clientcharm, "Tmate Client Charm", "", "Charm managing Tmate Client workload.")
+ }
+
+    System_Ext(tmateclient, "Tmate Client", "Client that shares his terminal session")
+    System_Ext(collaborator, "Collaborator", "Another client who wants access to the terminal")
+
+
+
+
+    Rel(tmatessh, clientcharm, "provides connection details")
+    Rel(clientcharm, tmateclient, "init client with connection details")
+
+    Rel(tmateclient, tmatessh, "init session via ssh")
+    Rel(collaborator, tmatessh, "join session via ssh")
+
+
+    UpdateRelStyle(tmatessh, clientcharm, $offsetY="30", $offsetX="-70")
+    UpdateRelStyle(clientcharm, tmateclient, $offsetY="80", $offsetX="150")
+    UpdateRelStyle(tmateclient, tmatessh, $offsetY="-30", $offsetX="10")
+    UpdateRelStyle(collaborator, tmatessh, $offsetY="-30", $offsetX="80")
+```
+
+
+## OCI images
+
+We use [Rockcraft](https://canonical-rockcraft.readthedocs-hosted.com/en/latest/) to build OCI Images for tmate-ssh-server. 
+The images are defined in [tmate-ssh-server_rock](https://github.com/canonical/tmate-ssh-server-operator/tree/main/tmate-ssh-server_rock).
+They are published to the [Github Container registry](https://github.com/canonical/tmate-ssh-server-operator/pkgs/container/tmate-ssh-server).
```

Comment:
I think it is published to the github container registry, but is pushed to charmhub!